torchinductor/config.py

```diff
@@ -56,7 +56,8 @@
 
 # config specific to codegen/cpp.pp
 class cpp:
-    threads = -1  # set to cpu_count()
+    threads = -1  # set to torch.get_num_threads()
+    specialize_num_threads = False  # whether to specialize on the number of threads
```

Comment:
Does this change hurt single threaded performance?

Reply:
@jansel In parallel, we can also work on adding thread number to one of the guards of torch dynamo. Thread number is similar to the shapes that can be specialized to get better perf. Not only the kernels in torch inductor, oneDNN kernels are also specialized with number of threads and would do recompilation if thread number changes. The difference from the shapes is that the thread number changes less frequently and assuming it static as default makes more sense. But before the dynamo guards support thread number guards, better we don't specialize it by default. What do you think?

Reply:
Perhaps we should specialize by default then.  The only use case I can think of for dynamically changing thread count is performance measurement.  For that use case people can just call `torchdynamo.reset()`

Reply:
We have the use case that dynamically identify the number of inference streams and thread number per inference stream for best inference throughput. Streams can be assigned with different number of threads. But I agree with you that specializing thread number by default should be fine. I have changed the code accordingly.